### PR TITLE
wgengine/netstack: add a missing refcount decrement after packet injection

### LINKED ISSUE #3762: netstack: leaking packet buffers
From a userspace tailscaled that was using 450 MB of memory:

heap pprof:
```
(pprof) top
Showing nodes accounting for 441.30MB, 98.57% of 447.70MB total
Dropped 48 nodes (cum <= 2.24MB)
Showing top 10 nodes out of 20
      flat  flat%   sum%        cum   cum%
  264.63MB 59.11% 59.11%   400.79MB 89.52%  inet.af/netstack/tcpip/stack.NewPacketBuffer
  117.03MB 26.14% 85.25%   117.03MB 26.14%  inet.af/netstack/tcpip/stack.glob..func1
      30MB  6.70% 91.95%   430.79MB 96.22%  tailscale.com/wgengine/netstack.(*Impl).injectInbound
   19.12MB  4.27% 96.22%    19.12MB  4.27%  inet.af/netstack/refsvfs2.Register
   10.51MB  2.35% 98.57%    10.51MB  2.35%  runtime.allocm
         0     0% 98.57%   430.79MB 96.22%  golang.zx2c4.com/wireguard/device.(*Peer).RoutineSequentialReceiver
         0     0% 98.57%    19.12MB  4.27%  inet.af/netstack/tcpip/stack.(*packetBufferRefs).InitRefs (inline)
         0     0% 98.57%     6.01MB  1.34%  runtime.mcall
         0     0% 98.57%     4.50MB  1.01%  runtime.mstart
         0     0% 98.57%     4.50MB  1.01%  runtime.mstart0
```

/cc @danderson @maisem

### PR DESCRIPTION
**Pull request by bradfitz**
Fixes #3762
Updates #3745 (probably fixes?)

### CODE REVIEW

File: wgengine/netstack/netstack.go [hunk 1]
```diff
@@ -508,6 +508,7 @@ func (ns *Impl) injectInbound(p *packet.Parsed, t *tstun.Wrapper) filter.Respons
 		Data: vv,
 	})
 	ns.linkEP.InjectInbound(pn, packetBuf)
+	packetBuf.DecRef()
```
Comment (maisem) [inline wgengine/netstack/netstack.go:511]: optional: would be nice to add tests for this

**Review by maisem — APPROVED** [review]

File: wgengine/netstack/netstack.go (hunk 1 above)
Comment (bradfitz) [reply wgengine/netstack/netstack.go:511]: yeah, still thinking about that. I think the integration tests somehow, but that seems slow & unreliable.

I'd like something tighter.

Comment (bradfitz) [reply wgengine/netstack/netstack.go:523]: PTAL

**Review by maisem — APPROVED** [review]

File: wgengine/netstack/netstack.go (hunk 1 above)
Comment (maisem) [reply wgengine/netstack/netstack.go:523]: Thanks!

**Pull request merged**